src/hooks/useNameDetails.test.ts

```diff
@@ -46,3 +47,389 @@ describe('useNameDetails', () => {
     expect(mockUseProfile).toHaveBeenCalledWith(expect.objectContaining({ enabled: false }))
   })
 })
+
+const mockUseNameDetailsConfig = {
+  'eth-available-name': { name: 'available.eth'},
+  'eth-emancipated-2ld': { name: 'eth-emanicipated.eth'},
+  'eth-unwrapped-2ld': { name: '',}
+}
+
+type MockUseNameDetailsType = keyof typeof mockUseNameDetailsConfig
+const mockUseNameDetailsTypes = Object.keys(mockUseNameDetailsConfig) as MockUseNameDetailsType[]
+
+const makeMockUseNameDetailsData = (type: MockUseNameDetailsType) => {
+  const config = mockUseNameDetailsConfig[type]
+  const {name} = config
+  return match(type)
+.with('eth-available-name', () => ({
+  "error": null,
+  "normalisedName": "available.eth",
+  "isValid": true,
+  "profile": {
+      "texts": [],
+      "coins": [],
+      "contentHash": null,
+      "abi": null,
+      "resolverAddress": "0x0000000000000000000000000000000000000000"
+  },
+  "isLoading": false,
+  "isCachedData": false,
+  "registrationStatus": "available",
+  "type": "name",
+  "isShort": false,
+  "is2LD": true,
+  "isETH": true,
+  "labelDataArray": [
```

Comment:
you can just spread the validation func result into this to remove some bloat here

Reply:
removed the mock from useNameDetails since it's not used at this time.